ui/src/app/cohort-search/demographics/demographics.component.tsx

```diff
@@ -0,0 +1,592 @@
+    let max = +maxAge;
+    let min = +minAge;
+    if (minOrMax === 'min' && min > +max) {
+      min = max;
+    } else if (minOrMax === 'max' && max < min) {
+      max = min;
+    }
+    const count = ageTypeNodes[ageType]
+      .filter(node => node.age >= min && node.age <= max)
+      .reduce((acc, node) => acc + node.count, 0);
+    this.setState({count, loading: false}, () => this.centerAgeCount());
+  }
+
+  get showPreview() {
+    const {selections, wizard} = this.props;
+    return !this.state.loading
+      && (selections && selections.length > 0)
+      && !(wizard.type === CriteriaType.AGE && serverConfigStore.getValue().enableCBAgeTypeOptions);
+  }
+
+  // Checks if form is in its initial state and if a count already exists before setting the total age count
+  get setTotalAge() {
+    const {count, maxAge, minAge} = this.state;
+    return minAge === defaultMinAge && maxAge === defaultMaxAge && !count;
+  }
+
+  render() {
+    const {selectedIds, wizard} = this.props;
+    const {ageType, calculating, count, loading, maxAge, minAge, nodes} = this.state;
+    const isAge = wizard.type === CriteriaType.AGE;
+    const calcDisabled = calculating || count !== null;
+    return loading
+      ? <div style={{textAlign: 'center'}}><Spinner style={{marginTop: '3rem'}}/></div>
+      : <React.Fragment>
+        {isAge
+          // Age slider with number inputs
+          ? <div style={styles.ageContainer}>
+            <div style={styles.ageLabel}>
+              Age Range
+            </div>
+            <div style={styles.sliderContainer}>
+              <input style={styles.ageInput}
+                type='number'
+                id='min-age'
+                min={defaultMinAge} max={maxAge}
+                value={minAge}
+                onBlur={() => this.onMinBlur()}
+                onChange={(e) => this.onMinChange(e.target.value)}/>
+              <div style={serverConfigStore.getValue().enableCBAgeTypeOptions
+                ? {...styles.slider, marginBottom: '0.75rem'}
+                : styles.slider}>
+                {serverConfigStore.getValue().enableCBAgeTypeOptions && <div ref={(el) => this.ageWrapper = el} id='count-wrapper'>
+                  {calculating
+                    ? <Spinner size={16}/>
+                    : <span style={styles.count} id='age-count'>
+                      {count.toLocaleString()}
+                    </span>
+                  }
+                </div>}
+                <Nouislider behaviour='drag'
+                  connect
+                  instanceRef={(slider) => this.onSliderInit(slider)}
+                  onChange={() => this.updateAgeSelection()}
+                  onSlide={(v) => this.onSliderUpdate(v)}
+                  range={{min: +defaultMinAge, max: +defaultMaxAge}}
+                  start={[+defaultMinAge, +defaultMaxAge]}
+                  step={1}/>
+              </div>
+              <input style={styles.ageInput}
```

Comment:
Could this be pulled out into a `NumberInput` component? I know we have a `TextInput` and this seems similar.

Reply:
There is a [`NumberInput`](https://github.com/all-of-us/workbench/blob/master/ui/src/app/components/inputs.tsx#L242) component but it actually returns a regular text input, even though it tries to set `type='number'`. I can file a follow-up ticket to fix that component (doesn't look like it's currently being used anywhere else).

Reply:
Yeah, lets follow up by fixing that. Sounds good.

Reply:
Filed [RW-4947](https://precisionmedicineinitiative.atlassian.net/browse/RW-4947)